qemu/tests/virtio_scsi_mq.py

```diff
@@ -110,129 +225,37 @@ def proc_interrupts_results(results, irqs_pattern):
-    output = session.cmd_output(get_dev_cmd)
-    system_dev = re.findall(r"/dev/[svh]d\w+(?=\d+)", output)[0]
-    dd_timeout = int(re.findall(r"\d+", extra_image_size)[0])
-    fill_cmd = ""
-    count = 0
-    for dev in re.split(r"\s+", output):
-        if not dev:
-            continue
-        if not re.findall(system_dev, dev):
-            fill_cmd += " dd of=%s if=/dev/urandom bs=1M " % dev
-            fill_cmd += "count=%s oflag=direct &" % dd_timeout
-            count += 1
-    if count != images_num:
-        test.error("Disks are not all show up in system. Output "
-                   "from the check command: %s" % output)
-    # As Bug 1177332 exists, mq is not supported completely.
-    # So don't considering performance currently, dd_timeout is longer.
-    dd_timeout = dd_timeout * images_num * 2
-    session.cmd(fill_cmd, timeout=dd_timeout)
-
-    dd_thread_num = count
-    while dd_thread_num:
-        time.sleep(5)
-        dd_thread_num = session.cmd_output("pgrep -x dd", timeout=dd_timeout)
-
-    error_context.context("Check the interrupt queues in guest", logging.info)
-    output = session.cmd_output(irq_check_cmd)
-    next_irq_results, cpu_list = proc_interrupts_results(output, irq_name)
-    logging.debug('The info of interrupters after testing :')
-    for irq_watch in next_irq_results.keys():
-        logging.debug('%s : %s %s' % (irq_watch, next_irq_results[irq_watch]['count'],
-                                      next_irq_results[irq_watch]['irq_des']))
-    irq_bit_map = 0
-    for irq_watch in next_irq_results.keys():
-        for index, count in enumerate(next_irq_results[irq_watch]["count"]):
-            if (int(count) - int(prev_irq_results[irq_watch]["count"][index])) > 0:
-                irq_bit_map |= 2 ** index
-
-    error_msg = ""
-    cpu_not_used = []
-    for index, cpu in enumerate(cpu_list):
-        if 2 ** index & irq_bit_map != 2 ** index:
-            cpu_not_used.append(cpu)
-
-    if cpu_not_used:
-        logging.debug("Interrupt info from procfs:\n%s" % output)
-        error_msg = " ".join(cpu_not_used)
-        if len(cpu_not_used) > 1:
-            error_msg += " are"
-        else:
-            error_msg += " is"
-        error_msg += " not used during test. Please check debug log for"
-        error_msg += " more information."
-        test.fail(error_msg)
+
+    error_context.context("Boot up guest with block devcie with num_queues"
+                          " is %s and smp is %s" % (num_queues, params['smp']),
+                          logging.info)
+    for vm in env.get_all_vms():
+        if vm.is_alive():
+            vm.destroy()
+    create_data_images()
+    params["start_vm"] = "yes"
+    vm = env.get_vm(params["main_vm"])
+    env_process.preprocess_vm(test, params, env, vm.name)
+    session = vm.wait_for_login(timeout=timeout)
+    if params.get("os_type") == "windows":
+        wmi_facility_test(session)
```

Comment:
Original 112 line "return" should be set out wmi_facility_test(), if set it in wmi_facility_test(), won't end the progress, check_irqbalance_status() also be executed.